screens/modals/RewardModal.js

```diff
@@ -1,175 +1,198 @@
+    this._updateState(rewardsAvailable, rewardsApplied, totalBalance);
+    this.setState({ isLoading: false });
+  }
+
+  // Forces a re-render when new props are passed
+  // TODO: this is deprecated - may need to find an alternative to getDerivedStateFromProps
+  componentWillReceiveProps(nextProps) {
+    const { rewardsAvailable, rewardsApplied, totalBalance } = nextProps;
+    if (this.state.totalBalance !== totalBalance) {
+      // Recalculate eligible rewards
+      this._updateState(rewardsAvailable, rewardsApplied, totalBalance);
     }
+  }
+
+  _updateState = (rewardsAvailable, rewardsApplied, totalBalance) => {
+    // Calculate eligible rewards
+    /* If negative balance exists, no additional rewards allowed!
+      Must take into account the current rewards applied
+     */
+    const additionalRewardsAllowed = totalBalance > 0 ? Math.ceil(totalBalance / rewardDollarValue) : 0;
+    const additionalRewardsAvailable = rewardsAvailable - rewardsApplied;
+    const additionalRewardsEligible = Math.min(additionalRewardsAllowed, additionalRewardsAvailable);
     this.setState({
+      rewardsAvailable,
       rewardsApplied,
-      rewardStatus,
-      totalPrice,
-      isLoading: false
+      totalBalance,
+      rewardsEligible: rewardsApplied + additionalRewardsEligible
     });
-  }
-
-  componentWillReceiveProps({ totalPrice }) {
-    this.setState(prevState => ({ ...prevState, totalPrice }));
-  }
+  };
 
   setModalVisible = visible => this.setState({ modalVisible: visible });
 
-  handleShowModal = () => {
-    const { totalPrice, rewardsAvailable, rewardsApplied } = this.props;
-    if ((totalPrice < 5 && rewardsApplied === 0) || rewardsAvailable === 0) {
-      return;
-    }
-    this.setModalVisible(!this.state.modalVisible);
+  showError = show => {
+    // Sets a 2s timeout for the showError
+    this.setState({ showError: show }, () => setTimeout(() => this.setState({ showError: false }), 2000));
   };
 
   handleApplyRewards = () => {
     // Communicate to parent component
-    this.props.callback(this.state.rewardsApplied, this.state.totalPrice);
+    this.props.callback(this.state.rewardsApplied, this.state.totalBalance);
     this.setModalVisible(!this.state.modalVisible);
   };
 
-  handleClear = () => {
-    const clearedStatus = this.state.rewardStatus.map(_ => false);
-    this.setState({
-      rewardsApplied: 0,
-      rewardStatus: clearedStatus,
-      totalPrice: this.state.totalPrice + 5 * this.state.rewardsApplied
-    });
-  };
-
-  updateReward = (i, apply) => {
-    if (apply && this.state.totalPrice < 5) {
-      return;
-    }
-
-    const rewardStatus = this.state.rewardStatus.slice();
-    rewardStatus[i] = !rewardStatus[i];
-
-    if (apply) {
-      this.setState({
-        rewardStatus,
-        rewardsApplied: this.state.rewardsApplied + 1,
-        totalPrice: this.state.totalPrice - 5
-      });
+  updateRewardsApplied = addToApplied => {
+    if (addToApplied) {
```

Comment:
Very minor detail but can we make the 'Max...' notif clear immediately if the minus button is pressed? It lingers a big which could be confusing. Can we add a `this.showError(false)` or something here so the error message immediately clears if they aren't at the max?